taxi_demo/machine_learning/tip_dask_xgboost.ipynb

```diff
@@ -0,0 +1,771 @@
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>...</th>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th></th>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th></th>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>\n",
+       "<div>Dask Name: sample, 19 tasks</div>"
+      ],
+      "text/plain": [
+       "Dask DataFrame Structure:\n",
+       "                    id pickup_taxizone_id dropoff_taxizone_id pickup_weekday pickup_weekofyear pickup_hour pickup_minute pickup_week_hour passenger_count tip_fraction\n",
+       "npartitions=19                                                                                                                                                        \n",
+       "                object            float64             float64          int64             int64       int64         int64            int64         float64      float64\n",
+       "                   ...                ...                 ...            ...               ...         ...           ...              ...             ...          ...\n",
+       "...                ...                ...                 ...            ...               ...         ...           ...              ...             ...          ...\n",
+       "                   ...                ...                 ...            ...               ...         ...           ...              ...             ...          ...\n",
+       "                   ...                ...                 ...            ...               ...         ...           ...              ...             ...          ...\n",
+       "Dask Name: sample, 19 tasks"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "sample.persist()"
```

Comment:
I believe `persist()` returns back the persisted object, so you would need to capture that back. Otherwise you wont actually be use the persisted object. Also best to wait for the persisting to finish so that way the subsequent operations are blocking

```python
from dask.distributed import wait
sample = sample.persist()
_ = wait(sample)  # wait returns a bunch of nonsense
```

Reply:
ah! And I actually did that in the benchmarks, but forgot. Ok I can update it, thank you

Reply:
this 100% could explain why my Dask code was slower, oy

Reply:
just pushed that change, thanks!